lib/rest/accounts/v1/credential/aws.js

```diff
@@ -66,34 +66,35 @@ AwsList = function AwsList(version) {
    * @memberof Twilio.Accounts.V1.CredentialContext.AwsList
    * @instance
    *
-   * @param {object|function} opts - ...
+   * @param {object} [opts] - ...
    * @param {number} [opts.limit] -
    *         Upper limit for the number of records to return.
    *         each() guarantees never to return more than limit.
    *         Default is no limit
-   * @param {number} [opts.pageSize=50] -
+   * @param {number} [opts.pageSize] -
    *         Number of records to fetch per request,
    *         when not set will use the default value of 50 records.
    *         If no pageSize is defined but a limit is defined,
    *         each() will attempt to read the limit with the most efficient
    *         page size, i.e. min(limit, 1000)
    * @param {Function} [opts.callback] -
    *         Function to process each record. If this and a positional
-   * callback are passed, this one will be used
+   *         callback are passed, this one will be used
    * @param {Function} [opts.done] -
    *          Function to be called upon completion of streaming
    * @param {Function} [callback] - Function to process each record
    */
   /* jshint ignore:end */
   AwsListInstance.each = function each(opts, callback) {
-    opts = opts || {};
     if (_.isFunction(opts)) {
-      opts = { callback: opts };
-    } else if (_.isFunction(callback) && !_.isFunction(opts.callback)) {
-      opts.callback = callback;
+      callback = opts;
+      opts = {};
```

Comment:
This logic seems flawed. I didn't try it but in my head if I run this:

```js
function someCallback(item) { console.log(item) }
each({ pageSize: 20 }, someCallback);
```

Then `opts` would be overridden in line 91 and therefore my options being dropped

Reply:
no, line 91 only executes if you omit the `opts` argument. I'll update the test case to include this.